Mono: Can't build master w/ latest compilation changes

**Godot version:**
Master

**OS/device including version:**
Windows 10 64bit

**Issue description:**
This https://github.com/godotengine/godot/commit/0deb8dda517aee11280bcaf12d96456d5a404ad3 is breaking mono building for me.

Error:
```
d:\godot-devel\build\godot\modules\mono\mono_gd\../utils/thread_local.h(79): error C2059: syntax error: '*'
d:\godot-devel\build\godot\modules\mono\mono_gd\../utils/thread_local.h(79): error C2091: function returns function
[Initial build] Compiling ==> modules\mono\godotsharp_dirs.cpp
godotsharp_dirs.cpp
[Initial build] Compiling ==> modules\mono\mono_gc_handle.cpp
mono_gc_handle.cpp
d:\godot-devel\build\godot\modules\mono\mono_gd\../utils/thread_local.h(79): error C2059: syntax error: '*'
d:\godot-devel\build\godot\modules\mono\mono_gd\../utils/thread_local.h(79): error C2091: function returns function
scons: *** [modules\mono\mono_gc_handle.windows.tools.64.obj] Error 2
scons: *** [modules\mono\csharp_script.windows.tools.64.obj] Error 2
scons: building terminated because of errors.
```

**Steps to reproduce:**
1. Open x64 Native Tools Command Prompt for VS 2017
2. Run `scons p=windows tools=yes module_mono_enabled=yes mono_glue=no -j3`
3. Wait for error

--- a/modules/mono/utils/thread_local.cpp
+++ b/modules/mono/utils/thread_local.cpp
@@ -69,7 +69,7 @@ struct ThreadLocalStorage::Impl {
 #define _CALLBACK_FUNC_
 #endif
 
-	Impl(void _CALLBACK_FUNC_ (*p_destr_callback_func)(void *)) {
+	Impl(void (_CALLBACK_FUNC_ *p_destr_callback_func)(void *)) {
 #ifdef WINDOWS_ENABLED
 		dwFlsIndex = FlsAlloc(p_destr_callback_func);
 		ERR_FAIL_COND(dwFlsIndex == FLS_OUT_OF_INDEXES);
@@ -95,7 +95,7 @@ void ThreadLocalStorage::set_value(void *p_value) const {
 	pimpl->set_value(p_value);
 }
 
-void ThreadLocalStorage::alloc(void _CALLBACK_FUNC_ (*p_destr_callback)(void *)) {
+void ThreadLocalStorage::alloc(void (_CALLBACK_FUNC_ *p_destr_callback)(void *)) {
 	pimpl = memnew(ThreadLocalStorage::Impl(p_destr_callback));
 }
 

--- a/modules/mono/utils/thread_local.h
+++ b/modules/mono/utils/thread_local.h
@@ -76,7 +76,7 @@ struct ThreadLocalStorage {
 	void *get_value() const;
 	void set_value(void *p_value) const;
 
-	void alloc(void _CALLBACK_FUNC_ (*p_dest_callback)(void *));
+	void alloc(void (_CALLBACK_FUNC_ *p_dest_callback)(void *));
 	void free();
 
 private:

